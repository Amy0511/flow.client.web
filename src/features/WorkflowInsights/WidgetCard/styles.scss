@import "~Styles/styles";

.c-widget-card {
  box-sizing: border-box;
  height: auto;
  min-width: 21.75rem;
  width: auto;
  min-height: 17.5rem;
  border-radius: 0.25rem;
  background-color: $cool-white-2;
  margin: 1rem;
}

.b-widget-card__title {
  min-height: 1.25rem;
  color: $gray-50;
  border-bottom: 0.0625rem solid $gray-20;
  text-transform: uppercase;
  text-align: center;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.25rem;
  position: relative;
  padding: 1.3125rem 0.625rem 0.625rem 0.625rem;
}

.b-widget-card__content {
  padding: 0.625rem;
  align-self: center;
}

.b-widget-card__value {
  display: flex;
  height: 13rem;
  width: 23rem;
  justify-content: center;
  align-items: center;
  color: $gray-90;
  font-size: 5rem;
  font-weight: 500;
  line-height: 6.5rem;
}
