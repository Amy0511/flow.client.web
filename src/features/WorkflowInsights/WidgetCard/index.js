import React from "react";
import PropTypes from "prop-types";
import "./styles.scss";

const WidgetCard = ({ title, children }) => {
  return (
    <div className="c-widget-card">
      <div className="b-widget-card__title">{title}</div>
      <div className="b-widget-card__content">{children}</div>
    </div>
  );
};

WidgetCard.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired
};

export default WidgetCard;
