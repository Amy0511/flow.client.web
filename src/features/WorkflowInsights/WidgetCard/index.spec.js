import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import WidgetCard from "./index";

const title = "test title";
const children = "test children";

describe("WidgetCard --- Snapshot", () => {
  it("Capturing Snapshot of WidgetCard", () => {
    const renderedValue = renderer.create(
      <MemoryRouter>
        <WidgetCard title={title} children={children} />
      </MemoryRouter>
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("WidgetCard --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <WidgetCard title={title} children={children} />
      </MemoryRouter>
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
