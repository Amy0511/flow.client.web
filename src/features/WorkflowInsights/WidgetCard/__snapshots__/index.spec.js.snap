// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetCard --- Snapshot Capturing Snapshot of WidgetCard 1`] = `
<div
  className="b-widget-card"
>
  <div
    className="b-widget-card__title"
  >
    test title
  </div>
  <div
    className="b-widget-card__content"
  >
    test children
  </div>
</div>
`;
